pkg/dyninst/procsubscribe/procscan/scanner.go

```diff
@@ -44,8 +45,11 @@ type Scanner struct {
 	// nowTicks returns the current time in ticks since boot.
 	nowTicks func() (ticks, error)
 
-	// live tracks discovered processes that have been reported as live.
-	live *btree.BTreeG[uint32]
+	mu struct {
+		sync.Mutex
+		// live tracks discovered processes that have been reported as live.
+		live *btree.BTreeG[uint32]
+	}
```

Comment:
The comment on the top would need to be updated if you made this thing thread safe, but I recommend you leave it as it was and delegate synchronization to the Subscriber